flytekit/extras/tasks/shell.py

```diff
@@ -237,3 +237,145 @@ def execute(self, **kwargs) -> typing.Any:
+            inputs=inputs,
+            output_locs=output_locs,
+            **kwargs,
+        )
+
+    def make_export_string_from_env_dict(self, d: typing.Dict[str, str]) -> str:
+        """
+        Utility function to convert a dictionary of desired environment variable key: value pairs into a string of
+        ```
+        export k1=v1
+        export k2=v2
+        ...
+        ```
+        """
+        items = []
+        for k, v in d.items():
+            items.append(f"export {k}={v}")
+        return "\n".join(items)
+
+    def execute(self, **kwargs) -> typing.Any:
+        """
+        Executes the given script by substituting the inputs and outputs and extracts the outputs from the filesystem
+        """
+        logger.info(f"Running shell script as type {self.task_type}")
+        if self.script_file:
+            with open(self.script_file) as f:
+                self._script = f.read()
+
+        outputs: typing.Dict[str, str] = {}
+        if self._output_locs:
+            for v in self._output_locs:
+                outputs[v.var] = self._interpolizer.interpolate(v.location, inputs=kwargs)
+
+        if os.name == "nt":
+            self._script = self._script.lstrip().rstrip().replace("\n", "&&")
+
+        if "env" in kwargs and isinstance(kwargs["env"], dict):
+            kwargs["export_env"] = self.make_export_string_from_env_dict(kwargs["env"])
+
+        gen_script = self._interpolizer.interpolate(self._script, inputs=kwargs, outputs=outputs)
+        if self._debug:
+            print("\n==============================================\n")
+            print(gen_script)
+            print("\n==============================================\n")
+
+        try:
+            subprocess.check_call(gen_script, shell=True)
+        except subprocess.CalledProcessError as e:
+            files = os.listdir(".")
+            fstr = "\n-".join(files)
+            logger.error(
+                f"Failed to Execute Script, return-code {e.returncode} \n"
+                f"StdErr: {e.stderr}\n"
+                f"StdOut: {e.stdout}\n"
+                f" Current directory contents: .\n-{fstr}"
+            )
+            raise
+
+        final_outputs = []
+        for v in self._output_locs:
+            if issubclass(v.var_type, FlyteFile):
+                final_outputs.append(FlyteFile(outputs[v.var]))
+            if issubclass(v.var_type, FlyteDirectory):
+                final_outputs.append(FlyteDirectory(outputs[v.var]))
+        if len(final_outputs) == 1:
+            return final_outputs[0]
+        if len(final_outputs) > 1:
+            return tuple(final_outputs)
+        return None
+
+
+# The raw_shell_task is an instance of _RawShellTask and wraps a 'pure' shell script
+# This utility function allows for the specification of env variables, arguments, and the actual script within the
+# workflow definition rather than at `_RawShellTask` instantiation
+def get_raw_shell_task(name: str = None) -> _RawShellTask:
```

Comment:
can we make name required? Else if you use the same function twice, then it would clash right? or does it fail?

Reply:
done

Reply:
> @myz540 firstly this is outstanding. Thank you for the patience. Just a couple more comments. I will work on this with priority. Lets merge after you handle couple of these comments.
> 
> So i understand you have an utility function to reduce boilerplate, but can we still use name. It is easy to make a mistake? But, you do more, you have a better usecase

@kumare3 I have updated the PR with your proposed changes and also run black, isort, and flake8 on the code prior to the github actions. It should be in a good state. My plan is open a PR into flytesnacks with some examples and further documentation but that doesn't necessarily need to block.